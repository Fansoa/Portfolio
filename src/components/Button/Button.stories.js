import Button from ".";

export default {
  title: "components/Button",
  component: Button,
  tags: ["autodocs"],
  args: {
    label: "Bouton",
    variantType: "fill",
  },
  argTypes: {
    className: {
      table: {
        disable: true,
      },
    },
    onClick: {
      action: "Button is clicked",
      description: "description",
    },
  },
};

export const Default = {};
