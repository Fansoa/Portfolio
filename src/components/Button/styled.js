import styled from "@emotion/styled";
import { pxToRem } from "../../theme/methods/pxToRem";

export const ButtonContainer = styled.button`
  ${({ theme, variantType }) => `
    cursor: pointer;
    user-select: none;
    padding: 0 ${pxToRem(10)} 0 ${pxToRem(10)};
    height: ${pxToRem(32)};
    font-size: ${pxToRem(12)};
    border-radius: ${pxToRem(5)};
    background-color: ${theme.buttonTheme[variantType].backgroundColor};
    border: 2px solid;
    border-color: ${theme.buttonTheme[variantType].borderColor};
  `}
`;
