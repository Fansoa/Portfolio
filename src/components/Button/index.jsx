import { ButtonContainer } from "./styled";
import PropTypes from 'prop-types'
import { noop } from "../../utils/methods/noop";

const Button = ({className, variantType, label, onClick}) => {
  return <ButtonContainer className={className} variantType={variantType} onClick={onClick}>{label}</ButtonContainer>;
};

Button.propTypes = {
  /**
   * Optional className
   */
  className: PropTypes.string,
  /**
   * Optional label
   */
  label: PropTypes.string,
  /**
   * Optional variantType
   */
  variantType: PropTypes.oneOf(['fill', 'stroke']),
  /**
   * Optional onClick
   */
  onClick: PropTypes.func
}

Button.defaultProps = {
  className: null,
  label: null,
  variantType: 'fill',
  onClick: noop,
}

export default Button;
