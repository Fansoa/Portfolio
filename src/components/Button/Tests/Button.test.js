import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "..";
import IntegrationProvider from "../../../../configs/jest/testUtils";

test("test Button", async () => {
  const result = render(
    <IntegrationProvider>
      <Button />
    </IntegrationProvider>
  );
  expect(result).toMatchSnapshot();
  screen.getByText("Toto");
});
