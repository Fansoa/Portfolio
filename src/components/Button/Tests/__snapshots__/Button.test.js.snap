// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  all: unset;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 5px 10px 5px 10px;
  border-radius: 5px;
  background-color: red;
}

<body>
    <div>
      <button
        class="emotion-0"
      >
        Label
      </button>
    </div>
  </body>,
  "container": .emotion-0 {
  all: unset;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 5px 10px 5px 10px;
  border-radius: 5px;
  background-color: red;
}

<div>
    <button
      class="emotion-0"
    >
      Label
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
